### ISSUE TITLE
[FEATURE]: Addition of a verification system

### ISSUE THREAD

**Issue by AbhijithGanesh** [pyrsia/pyrsia#394]
### Suggestion details

As the pyrsia project evolves, we'll be having more artifacts that operate within our ecosystem and we need to expand the scope for the same. To build trust around artifacts we have multiple approaches, some of them would be:
1.  A central node building the image through a pipeline ( Official images)
2. Verifying publications of various authors

We need an effective system that would allow the ecosystem to rate artifacts as verified images which then could be pulled by nodes that use them.

### Additional details

Docker Hub implements something like this through their [Docker Verified Publisher Program](https://docs.docker.com/docker-hub/publish/). We can take other sources of inspiration as well.

**Comment by erwin1** [comment]
First of all: I don't think this is an enhancement, but a real requirement.

What we’re after in the end is ‘trust’ that the artifact is the result of a build of a given (open) source.
Eventually Pyrsia nodes can do builds and come to a consensus which will be the basis of this trust.

But for now Pyrsia will need to rely on some other mechanism of trust, and the Docker Verified Publishers seems like a very good fit. However, there is currently no API for this at the Docker side. 

Alternatively we should choose to only support Official Images from Docker Hub, which are easily identifiable but quite limited in number.

A variation of this could be to keep a list of 'trusted' publishers somewhere in Pyrsia. 

To be discussed.

**Comment by AbhijithGanesh** [comment]
I completely agree with you. Initially we need to discuss what can be done for the MVP. Apart from that I think we need a system of our own because Pyrsia is more than Dockerhub and we need to come up with a logic that determines whether the publisher was a "verified publisher". 

PS: I don't know who assigns the labels 😅so I didn't know how enhancement got added. Probably because of the feature template

**Comment by prince-chrismc** [comment]
I fixed the label 😉 Probably the template added it.

I'd like to rename this issue, I think there's a really good question here...
_"What types of images are going to be included in MVP?"_
Is that alright by you?

Seems like the official are easiest (and probably the minimum?), perhaps the verified could be a secondary goal if a reasonable solution comes to light in time?

Also I think the concept of "verified publisher" in Pyrsia no just docker is a really interesting idea to explore!

**Comment by AbhijithGanesh** [comment]
Well how about this, if majority of the folks agree to the official image , we'll go for that in the MVP, then we'll come up with a standard that determines whether it's trusted or not. - This standard will be good for us as it'd be platform agnostic and it can easily be extended to other registries.

**Comment by AbhijithGanesh** [comment]
I found a [document](https://docs.docker.com/registry/spec/auth/oauth/) related to OAuth, Probably this should help us with the whole process.

**Comment by AbhijithGanesh** [comment]
API References:
 - Docker Hub [https://docs.docker.com/docker-hub/api/latest/]
 - Engine API [https://docs.docker.com/engine/api/]
